### ISSUE TITLE
change in block gas limit causes loss of consensus

### ISSUE THREAD

**Issue by svanegmond** [NethermindEth/nethermind#6100]
**Description**
While processing shyft mainnet chain from the beginning, nethermind 1.20.3, block processor loses consensus as soon as the block gas limit changed.

**Steps to Reproduce**

Sync Shyft chain from beginning. Any archive mode.   Nethermind [config template](https://github.com/shyftfederation/ansible-playbooks/blob/main/assets/nethermind.cfg.j2), [chainspec](https://spec.shyft.network/ShyftMainnet-current.json), [static nodes](https://spec.shyft.network/static-nodes-ipv4.json).

At block 4431021 per our chainspec there is a change to the block gas limit, identifying a contract that returns a constant.

Bug occurs in linux-arm64-1.20.3 official build from releases page.

Downgrading to 1.19.3 - no problem at this block. Processing continues normally.

**Actual behavior**

```
Processed     4430410...  4430777 |    705.72 ms  | slot   1,014 ms     | Gas gwei: 0.00 .. 0.00 (0.00) .. 0.00
- Blocks              1.37 MGas   |     65    txs | calls    368 (  0)  | sload  13,055 | sstore  1,840 | create   0
- Block throughput    1.93 MGas/s |     92.10 t/s |          521.45 b/s | recv        0 | proc        1
Block gas limit was changed from 10000000 to 40000000.
Processed     4430778...  4431022 |    694.12 ms  | slot   1,003 ms     | Gas gwei: 0.00 .. 0.00 (0.00) .. 0.00
- Blocks              0.99 MGas   |     47    txs | calls    245 (  0)  | sload   8,714 | sstore  1,234 | create   0
- Block throughput    1.42 MGas/s |     67.71 t/s |          352.96 b/s | recv        0 | proc       10
Processed block 4431025 (0x36b538...cb2871) is invalid:
- hash: expected 0xca517716bf5ea33db1af7a3a5cb667d62aa78a4b6bec3a09a2e04371b6a73145, got 0x36b5387aae28ede9eed2b6e688d8b0edb52628453b05ddeecd6bee80d0cb2871
- gas used: expected 50388, got 49344 (diff: -1044)
- receipts root: expected 0xc07bef762118c70cb5eb4800a3eaf3f370e0e700cec691be181475f2c503beef, got 0x5cac4c16400e8ebb88bb84cc65fed87d20b0d24a2ee04dd8d7333f476aa75142
- state root: expected 0x7bd01fdc0cb6b366fba26d896a3616997c2711bab70856bf6c34af47c3cc3ed2, got 0x7fe05cdef5fdc676a9b29b54215df5242ec8771feba3297f692183bcabc699e4
Processed block is not valid 4431025 (0xca517716bf5ea33db1af7a3a5cb667d62aa78a4b6bec3a09a2e04371b6a73145)
Suggested block TD: 1507799674885851355067799469895350738832750640, Suggested block IsPostMerge False, Block TD: 1507799674885851355067799469895350738832750640, Block >
Current contract validators (19): [0x61e952deeedd57e54a057fd5c612e7baf2dd5d09, 0x9435ca62d3b1341631fbe3d545e5e41ef9975c6f, 0xadd60cac91853ff3393e79af171027e38d03e9dc, >
Processed block 4431025 (0x36b538...cb2871) is invalid:
- hash: expected 0xca517716bf5ea33db1af7a3a5cb667d62aa78a4b6bec3a09a2e04371b6a73145, got 0x36b5387aae28ede9eed2b6e688d8b0edb52628453b05ddeecd6bee80d0cb2871
- gas used: expected 50388, got 49344 (diff: -1044)
- receipts root: expected 0xc07bef762118c70cb5eb4800a3eaf3f370e0e700cec691be181475f2c503beef, got 0x5cac4c16400e8ebb88bb84cc65fed87d20b0d24a2ee04dd8d7333f476aa75142
- state root: expected 0x7bd01fdc0cb6b366fba26d896a3616997c2711bab70856bf6c34af47c3cc3ed2, got 0x7fe05cdef5fdc676a9b29b54215df5242ec8771feba3297f692183bcabc699e4
Processed block is not valid 4431025 (0xca517716bf5ea33db1af7a3a5cb667d62aa78a4b6bec3a09a2e04371b6a73145)
Suggested block TD: 1507799674885851355067799469895350738832750640, Suggested block IsPostMerge False, Block TD: 1507799674885851355067799469895350738832750640, Block >
Created a Receipts trace of block 0xca517716bf5ea33db1af7a3a5cb667d62aa78a4b6bec3a09a2e04371b6a73145 in file /tmp/receipts_0xca517716bf5ea33db1af7a3a5cb667d62aa78a4b6b>
Deleting an invalid block or its descendant 0xb4949293aa18427d17a8d8a73a1f635b44699d2fb06c83a216668bcd67327b70

... etc, etc
```

**Comment by svanegmond** [comment]
We have observed this issue happening with other 1.19.3 installs, but only on sealers.  The ones not mining completed a sync successfully.

Downgraded my sealer to 1.18.2, also failed to handle the change in block reward.

Attempting anew with 1.17.

**Comment by svanegmond** [comment]
I have continued against this problem. The problem really makes no sense.   A tale of two machines. Both machines are running 1.20.3. The difference between the two machines is the chainspec they are running, they identify a fork that is due to happen in a few weeks.

This is the diff in chainspec:
` diff chainspec.json  /opt/nm/chainspec.json 
64c64,77
<         "eip2200Transition": "4431021"
---
>         "eip2200Transition": "4431021",
>         "eip1559Transition": "12721774",
>         "eip1884Transition": "12721774",
>         "eip2028Transition": "12721774",
>         "eip2200Transition": "12721774",
>         "eip2565Transition": "12721774",
>         "eip2929Transition": "12721774",
>         "eip2930Transition": "12721774",
>         "eip3198Transition": "12721774",
>         "eip3529Transition": "12721774",
>         "eip3541Transition": "12721774",
>         "eip3651Transition": "12721774",
>         "eip3855Transition": "12721774",
>         "eip3860Transition": "12721774"`

ie the entirety of the chainspec diff is regarding future enabling of features.

Machine A completes a full sync of the chain, whereas machine B does not (error log below)

If I revise the chainspec on machine B to match that of A, it will continue syncing the chain.  If I set the chainspec back (to re-enable the future fork), it will sync a few thousand blocks and then sooner or later:

`20571]: 18 Sep 14:58:25 | Processed     4486844...  4487127 |    928.27 ms  | slot   1,002 ms     | Gas gwei: 0.00 .. 0.00 (0.00) .. 0.00
Sep 18 14:58:25 sydney Nethermind.Runner[20571]: 18 Sep 14:58:25 | - Blocks              0.00 MGas   |      0    txs | calls    284 (  0)  | sload   9,940 | sstore  1,420 | create   0
Sep 18 14:58:25 sydney Nethermind.Runner[20571]: 18 Sep 14:58:25 | - Block throughput    0.00 MGas/s |      0.00 t/s |          305.94 b/s | recv    3,531 | proc    2,000
Sep 18 14:58:26 sydney Nethermind.Runner[20571]: 18 Sep 14:58:26 | Processed     4487128...  4487369 |    846.96 ms  | slot   1,006 ms     | Gas gwei: 0.00 .. 0.00 (0.00) .. 0.00
Sep 18 14:58:26 sydney Nethermind.Runner[20571]: 18 Sep 14:58:26 | - Blocks              0.00 MGas   |      0    txs | calls    242 (  0)  | sload   8,470 | sstore  1,210 | create   0
Sep 18 14:58:26 sydney Nethermind.Runner[20571]: 18 Sep 14:58:26 | - Block throughput    0.00 MGas/s |      0.00 t/s |          285.73 b/s | recv    3,289 | proc    2,000
Sep 18 14:58:26 sydney Nethermind.Runner[20571]: 18 Sep 14:58:26 | Syncing previously downloaded blocks from DB (partial offline mode until it finishes)
Sep 18 14:58:27 sydney Nethermind.Runner[20571]: 18 Sep 14:58:27 | Processed     4487370...  4487565 |    980.28 ms  | slot   1,039 ms     | Gas gwei: 0.00 .. 0.00 (0.00) .. 0.00
Sep 18 14:58:27 sydney Nethermind.Runner[20571]: 18 Sep 14:58:27 | - Blocks              0.00 MGas   |      0    txs | calls    196 (  0)  | sload   6,860 | sstore    980 | create   0
Sep 18 14:58:27 sydney Nethermind.Runner[20571]: 18 Sep 14:58:27 | - Block throughput    0.00 MGas/s |      0.00 t/s |          199.94 b/s | recv    3,093 | proc    2,000
Sep 18 14:58:27 sydney Nethermind.Runner[20571]: 18 Sep 14:58:27 | Processed block 4487790 (0xc89e4c...56abf7) is invalid:
Sep 18 14:58:27 sydney Nethermind.Runner[20571]: 18 Sep 14:58:27 | - hash: expected 0x1c1135f5d00b9d65bec4680dfc37c92ea54d0a49b0d436b7cb71145ba126f30a, got 0xc89e4c152b63d23a271c673e77b677ebe2f5ba6de220adb928f34fcef456abf7
Sep 18 14:58:27 sydney Nethermind.Runner[20571]: 18 Sep 14:58:27 | - gas used: expected 97826, got 73874 (diff: -23952)
Sep 18 14:58:27 sydney Nethermind.Runner[20571]: 18 Sep 14:58:27 | - receipts root: expected 0x945a5d27b61d5b81dbc6d59ef751a0dbe3d678580a17797898831e669182796d, got 0x674d4b593f57a62be9f76c2a33d2d5797044496f96882455f102eeb921e40903
Sep 18 14:58:27 sydney Nethermind.Runner[20571]: 18 Sep 14:58:27 | - state root: expected 0x748c431ad5c749f4c87de9f31649a43e46d3e27dd465ac16299c7863e73fbb02, got 0x3e49eee1932468f59f7c39c58577f8631f51a8936c4be0de51f4521bf0ca8284
Sep 18 14:58:27 sydney Nethermind.Runner[20571]: 18 Sep 14:58:27 | Processed block is not valid 4487790 (0x1c1135f5d00b9d65bec4680dfc37c92ea54d0a49b0d436b7cb71145ba126f30a)
Sep 18 14:58:27 sydney Nethermind.Runner[20571]: 18 Sep 14:58:27 | Suggested block TD: 1527115803444118426946297929486215061355936657, Suggested block IsPostMerge False, Block TD: 1527115803444118426946297929486215061355936657, Block IsPostMerge False
Sep 18 14:58:27 sydney Nethermind.Runner[20571]: 18 Sep 14:58:27 | Current contract validators (19): [0x61e952deeedd57e54a057fd5c612e7baf2dd5d09, 0x9435ca62d3b1341631fbe3d545e5e41ef9975c6f, 0xadd60cac91853ff3393e79af171027e38d03e9dc, 0x4d5ecb61f5f61c02416e80b0c15a559e1ff7f46d, 0x70aa75b4fe251fe7e8d19d50190a37009e1bee42, 0x5fa994c3177ef009f3fb751f1c0237b2a4e77fc9, 0x2bb7b2644e407863349dfdf8e0a94e1d6a50e12a, 0x4a3a7b11373617384f661b74047b6bd76b199fee, 0x633136f560958b912308359df5c4f76ae07e5129, 0xfb1eb3e6082ff20b4fde465f5576b5c46b5d619e, 0xeef0acc1a79a069ad8ec64015acbe8241e5464ff, 0x0cc5b7024b4796df8c317bb6f6e76fc7c7c2b8f0, 0x2d66cac1c852962ebc9f7ac7728b4fb4a7e2c16d, 0xf33dc521f4e12c63f206b0733f6927f1f62fc973, 0xe4f06e77b412f0c3444c4c4cbea300deb71f0bfc, 0xea31a89c81ac26f1180b8903b2e514c60b1f69ab, 0x007e21a3bd9eb8a8f001727f4c4130512052f6e6, 0xf6b4ba63431864971a4ef2e638cb43aaa7745eab, 0x122ff0b2f7c83e94c2eb58b2a4f440fe804799ef].
Sep 18 14:58:27 sydney Nethermind.Runner[20571]: 18 Sep 14:58:27 | Processed block 4487790 (0xc89e4c...56abf7) is invalid:
Sep 18 14:58:27 sydney Nethermind.Runner[20571]: 18 Sep 14:58:27 | - hash: expected 0x1c1135f5d00b9d65bec4680dfc37c92ea54d0a49b0d436b7cb71145ba126f30a, got 0xc89e4c152b63d23a271c673e77b677ebe2f5ba6de220adb928f34fcef456abf7
Sep 18 14:58:27 sydney Nethermind.Runner[20571]: 18 Sep 14:58:27 | - gas used: expected 97826, got 73874 (diff: -23952)
Sep 18 14:58:27 sydney Nethermind.Runner[20571]: 18 Sep 14:58:27 | - receipts root: expected 0x945a5d27b61d5b81dbc6d59ef751a0dbe3d678580a17797898831e669182796d, got 0x674d4b593f57a62be9f76c2a33d2d5797044496f96882455f102eeb921e40903
Sep 18 14:58:27 sydney Nethermind.Runner[20571]: 18 Sep 14:58:27 | - state root: expected 0x748c431ad5c749f4c87de9f31649a43e46d3e27dd465ac16299c7863e73fbb02, got 0x3e49eee1932468f59f7c39c58577f8631f51a8936c4be0de51f4521bf0ca8284
Sep 18 14:58:27 sydney Nethermind.Runner[20571]: 18 Sep 14:58:27 | Processed block is not valid 4487790 (0x1c1135f5d00b9d65bec4680dfc37c92ea54d0a49b0d436b7cb71145ba126f30a)
Sep 18 14:58:27 sydney Nethermind.Runner[20571]: 18 Sep 14:58:27 | Suggested block TD: 1527115803444118426946297929486215061355936657, Suggested block IsPostMerge False, Block TD: 1527115803444118426946297929486215061355936657, Block IsPostMerge False
Sep 18 14:58:27 sydney Nethermind.Runner[20571]: 18 Sep 14:58:27 | Created a Receipts trace of block 0x1c1135f5d00b9d65bec4680dfc37c92ea54d0a49b0d436b7cb71145ba126f30a in file /tmp/receipts_0x1c1135f5d00b9d65bec4680dfc37c92ea54d0a49b0d436b7cb71145ba126f30a.txt
Sep 18 14:58:27 sydney Nethermind.Runner[20571]: 18 Sep 14:58:27 | Deleting an invalid block or its descendant 0xe9d1ba814caad5a61295ff7c21e1437d314cf4a5f0ab0c1e18da0618924d0e45
Sep 18 14:58:27 sydney Nethermind.Runner[20571]: 18 Sep 14:58:27 | Deleting invalid block 0x1c1135f5d00b9d65bec4680dfc37c92ea54d0a49b0d436b7cb71145ba126f30a at level 4487790
Sep 18 14:58:27 sydney Nethermind.Runner[20571]: 18 Sep 14:58:27 | Deleting an invalid block or its descendant 0x2e1214941272a870ce755e56bea62042510f0129237ba97680a3fe2c5f0e873a
Sep 18 14:58:27 sydney Nethermind.Runner[20571]: 18 Sep 14:58:27 | Deleting invalid block 0xe9d1ba814caad5a61295ff7c21e1437d314cf4a5f0ab0c1e18da0618924d0e45 at level 4487791
Sep 18 14:58:27 sydney Nethermind.Runner[20571]: 18 Sep 14:58:27 | Deleting an invalid block or its descendant 0x57e75ce61c91c894824aea6be3494f7774d299c006f4d090f3fdb61ee01da0ca
Sep 18 14:58:27 sydney Nethermind.Runner[20571]: 18 Sep 14:58:27 | Deleting invalid block 0x2e1214941272a870ce755e56bea62042510f0129237ba97680a3fe2c5f0e873a at level 4487792
Sep 18 14:58:27 sydney Nethermind.Runner[20571]: 18 Sep 14:58:27 | Deleting an invalid block or its descendant 0xabf18ece26c9d2f73bd208feb7b1cf14e71fbc543e5f31488816f9dc40671fd4
Sep 18 14:58:27 sydney Nethermind.Runner[20571]: 18 Sep 14:58:27 | Deleting invalid block 0x57e75ce61c91c894824aea6be3494f7774d299c006f4d090f3fdb61ee01da0ca at level 4487793
Sep 18 14:58:27 sydney Nethermind.Runner[20571]: 18 Sep 14:58:27 | Deleting an invalid block or its descendant 0x963595ef51a01a6ef6b731b4b5852a33a617da59fb451bff83fa0f90bf7d6657
Sep 18 14:58:27 sydney Nethermind.Runner[20571]: 18 Sep 14:58:27 | Deleting invalid block 0xabf18ece26c9d2f73bd208feb7b1cf14e71fbc543e5f31488816f9dc40671fd4 at level 4487794
Sep 18 14:58:27 sydney Nethermind.Runner[20571]: 18 Sep 14:58:27 | Deleting an invalid block or its descendant 0xddbca29a40f59e24637b0deca01b93a1e9c0e193a50b9a2090ede5630758157b
Sep 18 14:58:27 sydney Nethermind.Runner[20571]: 18 Sep 14:58:27 | Deleting invalid block 0x963595ef51a01a6ef6b731b4b5852a33a617da59fb451bff83fa0f90bf7d6657 at level 4487795
Sep 18 14:58:27 sydney Nethermind.Runner[20571]: 18 Sep 14:58:27 | Deleting an invalid block or its descendant 0x997200d5b9a1468581a4a021801dec7e53293254f7e2a07adb80be09b8126049
Sep 18 14:58:27 sydney Nethermind.Runner[20571]: 18 Sep 14:58:27 | Deleting invalid block 0xddbca29a40f59e24637b0deca01b93a1e9c0e193a50b9a2090ede5630758157b at level 4487796
Sep 18 14:58:27 sydney Nethermind.Runner[20571]: 18 Sep 14:58:27 | Deleting an invalid block or its descendant 0x6b08fadb36e94217d88509bcc450e2065491199f823db143fbe26cd30b11c570
Sep 18 14:58:27 sydney Nethermind.Runner[20571]: 18 Sep 14:58:27 | Deleting invalid block 0x997200d5b9a1468581a4a021801dec7e53293254f7e2a07adb80be09b8126049 at level 4487797
Sep 18 14:58:27 sydney Nethermind.Runner[20571]: 18 Sep 14:58:27 | Deleting an invalid block or its descendant 0xf5eaffdaf884db1c36462946c68191ba4e5bf8b2486e9e3aa096ef082aa0c749
Sep 18 14:58:27 sydney Nethermind.Runner[20571]: 18 Sep 14:58:27 | Deleting invalid block 0x6b08fadb36e94217d88509bcc450e2065491199f823db143fbe26cd30b11c570 at level 4487798
Sep 18 14:58:27 sydney Nethermind.Runner[20571]: 18 Sep 14:58:27 | Deleting an invalid block or its descendant 0x35b8fcb4c4d2e0baef3dc58ee6a7f47df84314057d4aca1e3b1c22d8766de649
Sep 18 14:58:27 sydney Nethermind.Runner[20571]: 18 Sep 14:58:27 | Deleting invalid block 0xf5eaffdaf884db1c36462946c68191ba4e5bf8b2486e9e3aa096ef082aa0c749 at level 4487799
Sep 18 14:58:27 sydney Nethermind.Runner[20571]: 18 Sep 14:58:27 | Deleting an invalid block or its descendant 0x73b97c94f30ecd3ce48e6f30898e242ceca0df4cf04c16fb2fcaba6d19b0ae05
Sep 18 14:58:27 sydney Nethermind.Runner[20571]: 18 Sep 14:58:27 | Deleting invalid block 0x35b8fcb4c4d2e0baef3dc58ee6a7f47df84314057d4aca1e3b1c22d8766de649 at level 4487800
...
`
